### PR TITLE
fix: [DHIS2-11669][DHIS2-11746] save new enrollment event validation & statuses

### PR DESCRIPTION
**Pull request by simonadomnisoru**
- Trigger the error popup when the user clicks the "Complete" button in the new enrollment event form. [DHIS2-11669](https://jira.dhis2.org/browse/DHIS2-11669). 
- send the correct event status when the user clicks the "Save without competing"

### CODE REVIEW

File: src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js [hunk 1]
```diff
@@ -184,18 +185,17 @@ const getSaveHandler = (
...
-            const { onSave, itemId, id, calculatedFoundation } = this.props;
-            onSave(
-                itemId,
-                id,
-                calculatedFoundation,
-                saveType,
-            );
+            const { onSave, itemId, id, calculatedFoundation, warnings, errors } = this.props;
+            if (saveType === addEventSaveTypes.COMPLETE && ((errors && errors.length > 0) || (warnings && warnings.length > 0))) {
+                this.showMessagesPopup(saveType);
```
Comment (jasminenguyennn) [inline src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js:195]: hi @simonadomnisoru  Why do we only show the message when the save type is COMPLETE?
should it show if there is error and warning even when save type is SAVE_WITHOUT_COMPLETING?

Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js:195]: Hey, 
If the user clicks the button "Save without completing", we allowed for the event to be saved without validating the fields. The error popup will be shown only when the user clicks the "Complete" button. [DHIS2-11669](https://jira.dhis2.org/browse/DHIS2-11669) 
Thanks!

File: src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js [hunk 2]
```diff
@@ -184,18 +185,18 @@ const getSaveHandler = (
...
 
         handleSave = (saveType?: ?string) => {
-            const { onSave, itemId, id, calculatedFoundation } = this.props;
-            onSave(
-                itemId,
-                id,
-                calculatedFoundation,
-                saveType,
-            );
+            console.log('handleSave', saveType);
```
Comment (jasminenguyennn) [inline src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js:193]: is there any reason why we should keep this console?

Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js]: Good catch! I forgot to remove it. I updated the code. Thank you

**Review by jasminenguyennn — APPROVED** [review]

**Pull request merged**